### ISSUE TITLE
Saving a new puzzle does not check for existing name

### ISSUE THREAD

**Issue by philhanna** [philhanna/crossword#125]
### Steps to reproduce

1. Create a puzzle
2. Name it **test-puzzle**
3. Save it
4. Create another puzzle
5. Name it also **test-puzzle**
6. Save it

The old **test-puzzle** is overwritten.

The system should not allow a new puzzle to be saved with the name of an existing puzzle.

***Same is true with new grids***

**Comment by philhanna** [comment]
### Save logic

```
if grid already has a name:
    do save
else:
    prompt for name
    check existence
    if does not exist then:
        set name
        do save
    else:
        prompt for OK
        if OK then:
            set name
            do save
```
### Save as logic

```
if new grid (no name):
    prompt for new name
    check existence
    if does not exist:
        set name = new name
        do save
    else
        error message
else (existing grid):
    prompt for new name
    if name == new name:
        keep old name
        do save
    else:
        check existence
        if does not exist:
            keep old name
            do save
        else:
            prompt for OK
            if OK:
                do save
return
```

**Comment by philhanna** [comment]
Fixed by the fix for issue #124
